File: Python/codility/lesson4_FrogRiverOne.py
# you can write to stdout for debugging purposes, e.g.
# print("this is a debug message")

def solution(X, A):
    # write your code in Python 3.6
    S = set()
    count = len(A)
    for i in range(0, count):
        S.add(A[i])
        if(len(S) == X):
            return i
    return -1

File: Codility/Java/lesson4_FrogRiverOne.java
// you can also use imports, for example:
// import java.util.*;

// you can write to stdout for debugging purposes, e.g.
// System.out.println("this is a debug message");
import java.util.HashSet;
import java.util.Set;


class Solution {
    public int solution(int X, int[] A) {
        // write your code in Java SE 8
        Set<Integer> s = new HashSet<Integer>();
        
        for (int i = 0; i < A.length; i++) {
            s.add(A[i]);
            if (s.size() == X) {
                return i;
            }
        }
        return -1;
    }
}